### ISSUE TITLE
Mounting volumes times out a lot, seems to depend upon the time of day

### ISSUE THREAD

**Issue by mrdfuse** [Azure/acs-engine#2989]
**Is this a request for help?**:
Yes

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE

---

**What version of acs-engine?**:
0.13.1

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:59:24Z", GoVersion:"go1.9.3", Compil
er:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.7", GitCommit:"b30876a5539f09684ff9fde266fda10b37738c9c", GitTreeState:"clean", BuildDate:"2018-01-16T21:52:38Z", GoVersion:"go1.8.3", Compil
er:"gc", Platform:"linux/amd64"}
```

**What happened**:
We often observe that mounting a persistent volume claim times out a lot, and succeeds after a few minutes. But sometimes that can be 15 minutes, 2 hours or never. I notice when I do this after business hours it succeeds immediately. We have been noticing this behaviour for months, both on AKS and on ACS.

**What you expected to happen**:
That the volume mounts immediately, or fails consistently :)

**How to reproduce it** (as minimally and precisely as possible):
Difficult to say as it doesn't happen consistently.

**Anything else we need to know**:
`beta.kubernetes.io/instance-type: Standard_D2_v2`

```
kubectl describe po opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz -n ci
Name:           opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz
Namespace:      ci
Node:           k8s-agentpool1-31958223-2/10.240.0.4
Start Time:     Wed, 16 May 2018 16:18:59 +0200
Labels:         app=jenkins-static-slave
                pod-template-hash=1845773401
                release=opinionated-cricket
Annotations:    kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"ci","name":"opinionated-cricket-jenkins-static-slave-5d89cc7845","uid":"13
557959-...
Status:         Running
IP:             10.240.0.26
Controlled By:  ReplicaSet/opinionated-cricket-jenkins-static-slave-5d89cc7845
Containers:
  jenkins-static-slave:
    Container ID:  docker://e02b06432eafeb2867f902e94a4a5f7b96b7a8189a77f37de296482d344bf10f
    Image:         dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6
    Image ID:      docker-pullable://dela-docker.jfrog.io/dela-jenkins-static-slave@sha256:8fd435f6833ec7cac93b0feaaf47e7e1e5c6509c905c1d68b0b0a76dab35f00b
    Port:          <none>
    Args:
      c25f1aa880602a2caa20e601053db59891d5d9b0cfa40ad23291deae2514b430
      jenkins-slave-2
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 16 May 2018 16:32:21 +0200
      Finished:     Wed, 16 May 2018 16:32:22 +0200
    Ready:          False
    Restart Count:  7
    Environment:
      JENKINS_URL:        http://pilfering-wildebeest-jenkins-svc.ci
    Mounts:
      /home/jenkins/.gradle from jenkins-slave-gradle-pv (rw)
      /home/jenkins/.npm from jenkins-slave-npm-pv (rw)
      /var/run/docker.sock from docker-socket (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-p9vc8 (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  docker-socket:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/docker.sock
    HostPathType:
  jenkins-slave-gradle-pv:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  jenkins-slave-gradle-pvc-2
    ReadOnly:   false
  jenkins-slave-npm-pv:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  jenkins-slave-npm-pvc-2
    ReadOnly:   false
  default-token-p9vc8:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-p9vc8
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  kind=slave
Tolerations:     <none>
Events:
  Type     Reason                 Age                From                                Message
  ----     ------                 ----               ----                                -------
  Normal   Scheduled              13m                default-scheduler                   Successfully assigned opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz to k8s-agentpool1-31958223-2
  Normal   SuccessfulMountVolume  13m                kubelet, k8s-agentpool1-31958223-2  MountVolume.SetUp succeeded for volume "docker-socket"
  Normal   SuccessfulMountVolume  13m                kubelet, k8s-agentpool1-31958223-2  MountVolume.SetUp succeeded for volume "default-token-p9vc8"
  Normal   SuccessfulMountVolume  12m                kubelet, k8s-agentpool1-31958223-2  MountVolume.SetUp succeeded for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1"
  Warning  FailedMount            11m                kubelet, k8s-agentpool1-31958223-2  Unable to mount volumes for pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-00
0d3a3880c1)": timeout expired waiting for volumes to attach/mount for pod "ci"/"opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz". list of unattached/unmounted volumes=[jenkins-slave-npm-pv]
  Normal   SuccessfulMountVolume  11m                kubelet, k8s-agentpool1-31958223-2  MountVolume.SetUp succeeded for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1"
  Normal   Pulling                11m (x3 over 11m)  kubelet, k8s-agentpool1-31958223-2  pulling image "dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6"
  Normal   Pulled                 11m (x3 over 11m)  kubelet, k8s-agentpool1-31958223-2  Successfully pulled image "dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6"
  Normal   Created                11m (x3 over 11m)  kubelet, k8s-agentpool1-31958223-2  Created container
  Normal   Started                11m (x3 over 11m)  kubelet, k8s-agentpool1-31958223-2  Started container
  Warning  FailedSync             11m (x4 over 11m)  kubelet, k8s-agentpool1-31958223-2  Error syncing pod
  Warning  BackOff                3m (x35 over 11m)  kubelet, k8s-agentpool1-31958223-2  Back-off restarting failed container
```

```
---
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: azure-storage-class
  namespace: ci
  labels:
    kubernetes.io/cluster-service: "true"
  annotations:
    storageclass.beta.kubernetes.io/is-default-class: "true"
provisioner: kubernetes.io/azure-disk
parameters:
  storageaccounttype: Standard_LRS
  kind: Shared
  location: westeurope
  cachingmode: None  
  fsType: ext4
reclaimPolicy: Retain
```

kubelet log
```
I0516 14:18:59.411834    2317 kubelet.go:1837] SyncLoop (ADD, "api"): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)"
I0516 14:18:59.527794    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.528079    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "docker-socket" (UniqueName: "kubernetes.io/host-path/135aaf8b-5914-11e8-ad8e-000d3a3880c1-docker-socket") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.528170    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "default-token-p9vc8" (UniqueName: "kubernetes.io/secret/135aaf8b-5914-11e8-ad8e-000d3a3880c1-default-token-p9vc8") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.528265    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:18:59.528390    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:00.028346928 +0000 UTC (durationBeforeRetry 500ms). Error: Volume has not been added to the list of VolumesInUse in the node's volume status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (Unique
Name: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:18:59.530483    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:00.030450635 +0000 UTC (durationBeforeRetry 500ms). Error: Volume has not been added to the list of VolumesInUse in the node's volume status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (Unique
Name: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.628625    2317 reconciler.go:257] operationExecutor.MountVolume started for volume "default-token-p9vc8" (UniqueName: "kubernetes.io/secret/135aaf8b-5914-11e8-ad8e-000d3a3880c1-default-token-p9vc8") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.628722    2317 reconciler.go:257] operationExecutor.MountVolume started for volume "docker-socket" (UniqueName: "kubernetes.io/host-path/135aaf8b-5914-11e8-ad8e-000d3a3880c1-docker-socket") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.628788    2317 operation_generator.go:484] MountVolume.SetUp succeeded for volume "docker-socket" (UniqueName: "kubernetes.io/host-path/135aaf8b-5914-11e8-ad8e-000d3a3880c1-docker-socket") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.640336    2317 operation_generator.go:484] MountVolume.SetUp succeeded for volume "default-token-p9vc8" (UniqueName: "kubernetes.io/secret/135aaf8b-5914-11e8-ad8e-000d3a3880c1-default-token-p9vc8") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:18:59.784028    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:00.029745    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:00.033826    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:01.033804393 +0000 UTC (durationBeforeRetry 1s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:00.130470    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:00.134210    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:01.134184764 +0000 UTC (durationBeforeRetry 1s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:01.133794    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:01.134672    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:01.138843    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:03.138821772 +0000 UTC (durationBeforeRetry 2s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:01.139568    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:03.139543575 +0000 UTC (durationBeforeRetry 2s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:03.142183    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:03.142343    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:03.147411    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:07.147380986 +0000 UTC (durationBeforeRetry 4s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:03.149687    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:07.149669195 +0000 UTC (durationBeforeRetry 4s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:05.192594    2317 server.go:779] GET /stats/summary/: (183.909179ms) 200 [[Go-http-client/1.1] 10.240.0.76:48542]
I0516 14:19:07.161392    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:07.161476    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:07.166421    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:15.166395321 +0000 UTC (durationBeforeRetry 8s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:07.166528    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:15.166509221 +0000 UTC (durationBeforeRetry 8s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://
pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:09.807498    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:15.194321    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:15.194390    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:15.198622    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:31.198532167 +0000 UTC (durationBeforeRetry 16s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https:/
/pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:15.199660    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:19:31.199642571 +0000 UTC (durationBeforeRetry 16s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https:/
/pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:19.842443    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:29.858621    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:31.251152    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:31.251313    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:31.255058    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:20:03.255017125 +0000 UTC (durationBeforeRetry 32s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https:/
/pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:19:31.255525    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:20:03.255507627 +0000 UTC (durationBeforeRetry 32s). Error: Volume not attached according to node status for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https:/
/pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:19:39.880715    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:49.901755    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:19:59.941446    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:20:03.324514    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:20:03.324594    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
E0516 14:20:03.328377    2317 nestedpendingoperations.go:264] Operation for "\"kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd\"" failed. No retries permitted until 2018-05-16 14:21:07.328211776 +0000 UTC (durationBeforeRetry 1m4s). Error: Volume not attached according to node status for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https:
//pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:20:03.329087    2317 operation_generator.go:684] Controller attach succeeded for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") device path: "2"
I0516 14:20:03.424940    2317 reconciler.go:257] operationExecutor.MountVolume started for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:20:03.425007    2317 operation_generator.go:416] MountVolume.WaitForAttach entering for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") DevicePath "2"
I0516 14:20:05.315878    2317 server.go:779] GET /stats/summary/: (284.369149ms) 200 [[Go-http-client/1.1] 10.240.0.76:48542]
I0516 14:20:06.363237    2317 azure_common_linux.go:194] azureDisk - Disk "/dev/disk/azure/scsi1/lun2" appears to be unformatted, attempting to format as type: "ext4" with options: [-E lazy_itable_init=0,lazy_journal_init=0 -F /dev/disk/azure/scsi1/lun2]
I0516 14:20:07.375716    2317 azure_common_linux.go:199] azureDisk - Disk successfully formatted (mkfs): ext4 - /dev/disk/azure/scsi1/lun2 tt
W0516 14:20:07.375746    2317 azure_common_linux.go:201] azureDisk - format of disk "/dev/disk/azure/scsi1/lun2" failed: type:("ext4") target:("tt") options:("o")error:(<nil>)
I0516 14:20:07.375782    2317 operation_generator.go:425] MountVolume.WaitForAttach succeeded for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:20:07.596862    2317 operation_generator.go:446] MountVolume.MountDevice succeeded for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") device mount path "/var/lib/kubelet/plugins/kubernetes.io/azure-
disk/mounts/b3835638752"
I0516 14:20:07.615058    2317 azure_mounter.go:162] azureDisk - successfully mounted disk dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1 on /var/lib/kubelet/pods/135aaf8b-5914-11e8-ad8e-000d3a3880c1/volumes/kubernetes.io~azure-disk/pvc-1342af71-5914-11e8-ad8e-000d3a3880c1
I0516 14:20:07.615841    2317 operation_generator.go:484] MountVolume.SetUp succeeded for volume "pvc-1342af71-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1342af71-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:20:09.957677    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:20:19.978595    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:20:29.995711    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:20:39.141668    2317 reconciler.go:407] Reconciler sync states: could not find pod information in desired or actual states or pending operation, update it in both states: &{volumeName:kubernetes.io/azure-disk/b3835638752 podName:135aaf8b-5914-11e8-ad8e-000d3a3880c1 volumeSpec:0xc420f3c140 outerVolumeSpecName:pvc-1342af71-5914-11e8-ad8e-000d3a3880c1 pod:0xc42275b180 pluginIsAttachable:true volumeGidValue: devicePath: reportedInUse:false mounter:0xc4234bd000}
I0516 14:20:39.148329    2317 reconciler.go:547] Volume: kubernetes.io/azure-disk/b3835638752 is mounted
I0516 14:20:40.012031    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:20:50.029116    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:21:00.057488    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
E0516 14:21:02.413357    2317 kubelet.go:1628] Unable to mount volumes for pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)": timeout expired waiting for volumes to attach/mount for pod "ci"/"opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz". list of unattached/unmounted volumes=[jenkins-slave-npm-pv]; skipping pod
E0516 14:21:02.413387    2317 pod_workers.go:182] Error syncing pod 135aaf8b-5914-11e8-ad8e-000d3a3880c1 ("opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)"), skipping: timeout expired waiting for volumes to attach/mount for pod "ci"/"opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz". list of unattached/unmounted volumes=[jenkins-slave-npm-pv]
I0516 14:21:05.182104    2317 server.go:779] GET /stats/summary/: (161.103794ms) 200 [[Go-http-client/1.1] 10.240.0.76:48542]
I0516 14:21:07.339977    2317 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:21:07.343924    2317 operation_generator.go:684] Controller attach succeeded for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") device path: "3"
I0516 14:21:07.440489    2317 reconciler.go:257] operationExecutor.MountVolume started for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:21:07.440578    2317 operation_generator.go:416] MountVolume.WaitForAttach entering for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") DevicePath "3"
I0516 14:21:10.075000    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:21:10.352227    2317 azure_common_linux.go:194] azureDisk - Disk "/dev/disk/azure/scsi1/lun3" appears to be unformatted, attempting to format as type: "ext4" with options: [-E lazy_itable_init=0,lazy_journal_init=0 -F /dev/disk/azure/scsi1/lun3]
I0516 14:21:11.515228    2317 azure_common_linux.go:199] azureDisk - Disk successfully formatted (mkfs): ext4 - /dev/disk/azure/scsi1/lun3 tt
W0516 14:21:11.515300    2317 azure_common_linux.go:201] azureDisk - format of disk "/dev/disk/azure/scsi1/lun3" failed: type:("ext4") target:("tt") options:("o")error:(<nil>)
I0516 14:21:11.516461    2317 operation_generator.go:425] MountVolume.WaitForAttach succeeded for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:21:11.982201    2317 operation_generator.go:446] MountVolume.MountDevice succeeded for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1") device mount path "/var/lib/kubelet/plugins/kubernetes.io/azure-
disk/mounts/b879764857"
I0516 14:21:12.030003    2317 azure_mounter.go:162] azureDisk - successfully mounted disk dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1 on /var/lib/kubelet/pods/135aaf8b-5914-11e8-ad8e-000d3a3880c1/volumes/kubernetes.io~azure-disk/pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1
I0516 14:21:12.030347    2317 operation_generator.go:484] MountVolume.SetUp succeeded for volume "pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1" (UniqueName: "kubernetes.io/azure-disk/https://pvc1001418384002.blob.core.windows.net/1001418384/dela-acs-dev-dynamic-pvc-1351aa76-5914-11e8-ad8e-000d3a3880c1.vhd") pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz" (UID: "135aaf8b-5914-11e8-ad8e-000d3a3880c1")
I0516 14:21:17.704172    2317 kuberuntime_manager.go:374] No sandbox for pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)" can be found. Need to start a new one
I0516 14:21:18.563633    2317 provider.go:119] Refreshing cache for provider: *credentialprovider.defaultDockerConfigProvider
I0516 14:21:18.564441    2317 provider.go:119] Refreshing cache for provider: *azure.acrProvider
I0516 14:21:18.863637    2317 kubelet.go:1871] SyncLoop (PLEG): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)", event: &pleg.PodLifecycleEvent{ID:"135aaf8b-5914-11e8-ad8e-000d3a3880c1", Type:"ContainerStarted", Data:"ae76ffbe106ea65edcd1a16f9bc9ea1994f5211b06748d40996429b714bc1943"}
I0516 14:21:19.834711    2317 kube_docker_client.go:333] Stop pulling image "dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6": "Status: Image is up to date for dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6"
I0516 14:21:20.100834    2317 kubelet_node_status.go:503] Using Node Hostname from cloudprovider: "k8s-agentpool1-31958223-2"
I0516 14:21:20.888560    2317 kubelet.go:1871] SyncLoop (PLEG): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)", event: &pleg.PodLifecycleEvent{ID:"135aaf8b-5914-11e8-ad8e-000d3a3880c1", Type:"ContainerStarted", Data:"03b1787f2f27d234186cad23dbd28c33746bb93ccc033f506f3d9db6e639eb0d"}
I0516 14:21:21.904037    2317 kubelet.go:1871] SyncLoop (PLEG): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)", event: &pleg.PodLifecycleEvent{ID:"135aaf8b-5914-11e8-ad8e-000d3a3880c1", Type:"ContainerDied", Data:"03b1787f2f27d234186cad23dbd28c33746bb93ccc033f506f3d9db6e639eb0d"}
I0516 14:21:22.207604    2317 kuberuntime_manager.go:503] Container {Name:jenkins-static-slave Image:dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6 Command:[] Args:[c25f1aa880602a2caa20e601053db59891d5d9b0cfa40ad23291deae2514b430 jenkins-slave-2] WorkingDir: Ports:[] EnvFrom:[] Env:[{Name:JENKINS_URL Value:http://pilfering-wildebeest-jenkins-svc.ci ValueFrom:nil} {Name:SONARQUBE_LOGIN Value:xxx ValueFrom:nil} {Name:SONARQUBE_HOST Value:ht
tps://sonar.delanet.be ValueFrom:nil} {Name:dockerRegistry Value:dela-docker.jfrog.io ValueFrom:nil} {Name:dockerImagePrefix Value:dela- ValueFrom:nil}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:docker-socket ReadOnly:false MountPath:/var/run/docker.sock SubPath: MountPropagation:<nil>} {Name:jenkins-slave-gradle-pv ReadOnly:false MountPath:/home/jenkins/.gradle SubPath: MountPropagation:<nil>} {Name:jenkins-slave-npm-pv ReadOnly:false MountPath:/home/jenkins/.np
m SubPath: MountPropagation:<nil>} {Name:default-token-p9vc8 ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount SubPath: MountPropagation:<nil>}] LivenessProbe:nil ReadinessProbe:nil Lifecycle:nil TerminationMessagePath:/dev/termination-log TerminationMessagePolicy:File ImagePullPolicy:Always SecurityContext:nil Stdin:false StdinOnce:false TTY:false} is dead, but RestartPolicy says that we should restart it.
I0516 14:21:22.207725    2317 kuberuntime_manager.go:742] checking backoff for container "jenkins-static-slave" in pod "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)"
I0516 14:21:22.670854    2317 kube_docker_client.go:333] Stop pulling image "dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6": "Status: Image is up to date for dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6"
I0516 14:21:23.940660    2317 kubelet.go:1871] SyncLoop (PLEG): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)", event: &pleg.PodLifecycleEvent{ID:"135aaf8b-5914-11e8-ad8e-000d3a3880c1", Type:"ContainerStarted", Data:"82102111df23da29c5090dee099b642e56befbfdc7c0f658885e76ff3f1ab31d"}
I0516 14:21:24.960235    2317 kubelet.go:1871] SyncLoop (PLEG): "opinionated-cricket-jenkins-static-slave-5d89cc7845-bd8kz_ci(135aaf8b-5914-11e8-ad8e-000d3a3880c1)", event: &pleg.PodLifecycleEvent{ID:"135aaf8b-5914-11e8-ad8e-000d3a3880c1", Type:"ContainerDied", Data:"82102111df23da29c5090dee099b642e56befbfdc7c0f658885e76ff3f1ab31d"}
I0516 14:21:25.264084    2317 kuberuntime_manager.go:503] Container {Name:jenkins-static-slave Image:dela-docker.jfrog.io/dela-jenkins-static-slave:1.0.6 Command:[] Args:[c25f1aa880602a2caa20e601053db59891d5d9b0cfa40ad23291deae2514b430 jenkins-slave-2] WorkingDir: Ports:[] EnvFrom:[] Env:[{Name:JENKINS_URL Value:http://pilfering-wildebeest-jenkins-svc.ci ValueFrom:nil} {Name:SONARQUBE_LOGIN Value:xxx ValueFrom:nil} {Name:SONARQUBE_HOST Value:ht
tps://sonar.delanet.be ValueFrom:nil} {Name:dockerRegistry Value:dela-docker.jfrog.io ValueFrom:nil} {Name:dockerImagePrefix Value:dela- ValueFrom:nil}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:docker-socket ReadOnly:false MountPath:/var/run/docker.sock SubPath: MountPropagation:<nil>} {Name:jenkins-slave-gradle-pv ReadOnly:false MountPath:/home/jenkins/.gradle SubPath: MountPropagation:<nil>} {Name:jenkins-slave-npm-pv ReadOnly:false MountPath:/home/jenkins/.np
```

Maybe these lines are relevant?
```
I0516 14:21:10.352227    2317 azure_common_linux.go:194] azureDisk - Disk "/dev/disk/azure/scsi1/lun3" appears to be unformatted, attempting to format as type: "ext4" with options: [-E lazy_itable_init=0,lazy_journal_init=0 -F /dev/disk/azure/scsi1/lun3]
I0516 14:21:11.515228    2317 azure_common_linux.go:199] azureDisk - Disk successfully formatted (mkfs): ext4 - /dev/disk/azure/scsi1/lun3 tt
W0516 14:21:11.515300    2317 azure_common_linux.go:201] azureDisk - format of disk "/dev/disk/azure/scsi1/lun3" failed: type:("ext4") target:("tt") options:("o")error:(<nil>)

**Comment by andyzhangx** [comment]
could you share your helm chart and config if possible?

**Comment by andyzhangx** [comment]
And also when the azuredisk mount on a pod never succeeded, could you check:
 - whether that disk has already been attached to the agent VM?
 - run `kubectl get no NODE-NAME -o yaml > node.log` and paste `volumesInUse` and `volumesAttached` list on that VM

**Comment by mrdfuse** [comment]
I think these are the most relevant files from the chart (happens with any chart, or even before we started using Helm btw):
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: {{ template "jenkins-static-slave.fullname" . }}
  namespace: {{ .Values.namespace }}
  labels:
    app: {{ template "jenkins-static-slave.name" . }}
    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
spec:
  replicas: {{ .Values.replicaCount }}
  template:
    metadata:
      labels:
        app: {{ template "jenkins-static-slave.name" . }}
        release: {{ .Release.Name }}
    spec: 
      securityContext:
        runAsUser: 10000
        fsGroup: 10000 
      imagePullSecrets:
        - name: {{ .Values.imagePullSecret }}
      nodeSelector:
        kind: {{ .Values.nodeSelector }}
      containers:
        - name: {{ .Chart.Name }}
          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
          imagePullPolicy: {{ .Values.image.pullPolicy }}
          args: ["{{ .Values.slaveToken }}", "jenkins-slave-{{ .Values.slaveIndex }}"]
          env:
            - name: JENKINS_URL
              value: {{ .Values.jenkinsUrl }}
            - name: dockerRegistry
              value: dela-docker.jfrog.io
            - name: dockerImagePrefix
              value: dela-
          volumeMounts:
            - name: docker-socket
              mountPath: /var/run/docker.sock
            - name: jenkins-slave-gradle-pv
              mountPath: /home/jenkins/.gradle
            - name: jenkins-slave-npm-pv
              mountPath: /home/jenkins/.npm
      volumes:
      - name: docker-socket
        hostPath:
          path: /var/run/docker.sock
      - name: jenkins-slave-gradle-pv
        persistentVolumeClaim:
          claimName: jenkins-slave-gradle-pvc-{{ .Values.slaveIndex }}
      - name: jenkins-slave-npm-pv
        persistentVolumeClaim:
          claimName: jenkins-slave-npm-pvc-{{ .Values.slaveIndex }}
```

```
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: jenkins-slave-gradle-pvc-{{ .Values.slaveIndex }}
  namespace: ci
spec:
  storageClassName: azure-storage-class
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
```

```
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: jenkins-slave-npm-pvc-{{ .Values.slaveIndex }}
  namespace: ci
spec:
  storageClassName: azure-storage-class
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
```

I have shared the storage class definition in my previous comment.

As for the other requests, I'll provide the info as we encounter the problem again. For now it seems work ok again (few minutes waiting time)

**Comment by neurot1cal** [comment]
@andyzhangx Do you know how long we can expect a normal PV mount to take? Just spun up a branc new 1.11.2 cluster using latest acs-engine and the pod never comes up because the pod has unbound PersistentVolumeClaims;

```
 helm --namespace jenkins --name jenkins -f jenkins-values.yaml install stable/jenkins

NAME:   jenkins
LAST DEPLOYED: Tue Aug 14 16:30:34 2018
NAMESPACE: jenkins
STATUS: DEPLOYED

RESOURCES:
==> v1/Secret
NAME     TYPE    DATA  AGE
jenkins  Opaque  2     1s

==> v1/ConfigMap
NAME           DATA  AGE
jenkins        4     1s
jenkins-tests  1     1s

==> v1beta1/ClusterRoleBinding
NAME                  AGE
jenkins-role-binding  1s

==> v1/PersistentVolumeClaim
NAME     STATUS   VOLUME   CAPACITY  ACCESS MODES  STORAGECLASS  AGE
jenkins  Pending  default  1s

==> v1/ServiceAccount
NAME     SECRETS  AGE
jenkins  1        1s

==> v1/Service
NAME           TYPE       CLUSTER-IP    EXTERNAL-IP  PORT(S)    AGE
jenkins-agent  ClusterIP  10.0.103.103  <none>       50000/TCP  1s
jenkins        ClusterIP  10.0.45.179   <none>       8080/TCP   1s

==> v1beta1/Deployment
NAME     DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
jenkins  1        1        1           0          1s

==> v1/NetworkPolicy
NAME                    POD-SELECTOR                        AGE
jenkins-jenkins-master  component=jenkins-jenkins-master    1s
jenkins-jenkins-slave   jenkins/jenkins-jenkins-slave=true  1s

==> v1/Pod(related)
NAME                      READY  STATUS   RESTARTS  AGE
jenkins-78df46c4f9-n2xrk  0/1    Pending  0         1s

 k get all -n jenkins
NAME                           READY     STATUS     RESTARTS   AGE
pod/jenkins-78df46c4f9-n2xrk   0/1       Init:0/1   3          12m

NAME                    TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)     AGE
service/jenkins         ClusterIP   10.0.45.179    <none>        8080/TCP    12m
service/jenkins-agent   ClusterIP   10.0.103.103   <none>        50000/TCP   12m

NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/jenkins   1         1         1            0           12m

NAME                                 DESIRED   CURRENT   READY     AGE
replicaset.apps/jenkins-78df46c4f9   1         1         0         12m

 k describe pod/jenkins-78df46c4f9-n2xrk -n jenkins
Name:               jenkins-78df46c4f9-n2xrk
Namespace:          jenkins
Priority:           0
PriorityClassName:  <none>
Node:               k8s-linuxpool2-83938631-vmss000001/10.240.0.35
Start Time:         Tue, 14 Aug 2018 16:30:41 -0700
Labels:             app=jenkins
                    chart=jenkins-0.16.1
                    component=jenkins-jenkins-master
                    heritage=Tiller
                    pod-template-hash=3489027095
                    release=jenkins
Annotations:        checksum/config=3242ce9d4b04da8935cf38349d1dc0de28402f13522bf2ac16640501c5ae4028
Status:             Pending
IP:                 10.240.0.50
Controlled By:      ReplicaSet/jenkins-78df46c4f9
Init Containers:
  copy-default-config:
    Container ID:  docker://f57f8a533b3f2c1b2d4d32b200a5cd4cfa6b26d0905c3707cb1d10b5943f164e
    Image:         jenkins/jenkins:2.121.2-slim
    Image ID:      docker-pullable://jenkins/jenkins@sha256:8f7474fc19d85de7d03d26e34591942a500b9722a900c987887d4cb9c332c11b
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      /var/jenkins_config/apply_config.sh
    State:          Running
      Started:      Tue, 14 Aug 2018 16:40:46 -0700
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Tue, 14 Aug 2018 16:37:26 -0700
      Finished:     Tue, 14 Aug 2018 16:40:19 -0700
    Ready:          False
    Restart Count:  3
    Environment:    <none>
    Mounts:
      /usr/share/jenkins/ref/secrets/ from secrets-dir (rw)
      /var/jenkins_config from jenkins-config (rw)
      /var/jenkins_home from jenkins-home (rw)
      /var/jenkins_plugins from plugin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from jenkins-token-f95v7 (ro)
Containers:
  jenkins:
    Container ID:
    Image:         jenkins/jenkins:2.121.2-slim
    Image ID:
    Ports:         8080/TCP, 50000/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      --argumentsRealm.passwd.$(ADMIN_USER)=$(ADMIN_PASSWORD)
      --argumentsRealm.roles.$(ADMIN_USER)=admin
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:      1
      memory:   1Gi
    Liveness:   http-get http://:http/login delay=60s timeout=5s period=10s #success=1 #failure=12
    Readiness:  http-get http://:http/login delay=60s timeout=1s period=10s #success=1 #failure=3
    Environment:
      JAVA_OPTS:       -Xms512m -Xmx512m
      JENKINS_OPTS:
      ADMIN_PASSWORD:  <set to the key 'jenkins-admin-password' in secret 'jenkins'>  Optional: false
      ADMIN_USER:      <set to the key 'jenkins-admin-user' in secret 'jenkins'>      Optional: false
    Mounts:
      /usr/share/jenkins/ref/plugins/ from plugin-dir (rw)
      /usr/share/jenkins/ref/secrets/ from secrets-dir (rw)
      /var/jenkins_config from jenkins-config (ro)
      /var/jenkins_home from jenkins-home (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from jenkins-token-f95v7 (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  jenkins-config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      jenkins
    Optional:  false
  plugin-dir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
  secrets-dir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
  jenkins-home:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  jenkins
    ReadOnly:   false
  jenkins-token-f95v7:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  jenkins-token-f95v7
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  Role=Utility
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age                From                                         Message
  ----     ------                  ----               ----                                         -------
  Warning  FailedScheduling        12m (x7 over 12m)  default-scheduler                            pod has unbound PersistentVolumeClaims (repeated 3 times)
  Normal   Scheduled               12m                default-scheduler                            Successfully assigned jenkins/jenkins-78df46c4f9-n2xrk to k8s-linuxpool2-83938631-vmss000001
  Normal   SuccessfulAttachVolume  12m                attachdetach-controller                      AttachVolume.Attach succeeded for volume "pvc-0bebabba-a01a-11e8-81ab-000d3a377cde"
  Warning  BackOff                 2m (x3 over 5m)    kubelet, k8s-linuxpool2-83938631-vmss000001  Back-off restarting failed container
  Normal   Pulling                 2m (x4 over 11m)   kubelet, k8s-linuxpool2-83938631-vmss000001  pulling image "jenkins/jenkins:2.121.2-slim"
  Normal   Pulled                  2m (x4 over 11m)   kubelet, k8s-linuxpool2-83938631-vmss000001  Successfully pulled image "jenkins/jenkins:2.121.2-slim"
  Normal   Created                 2m (x4 over 11m)   kubelet, k8s-linuxpool2-83938631-vmss000001  Created container
  Normal   Started                 2m (x4 over 11m)   kubelet, k8s-linuxpool2-83938631-vmss000001  Started container
```

Any ideas?

**Comment by andyzhangx** [comment]
@neurot1cal The disk mount time cost is expected to complete within 1 min, while your case is the PVC disk creation costs a lot of time. You may run "kubectl describe PVC jenkins" to get more details.

**Comment by neurot1cal** [comment]
@andyzhangx seems to be fine; 

```
☁  jenkins [cluster-lifecycle] ⚡ kubectl describe pvc -n jenkins
Name:          jenkins
Namespace:     jenkins
StorageClass:  default
Status:        Bound
Volume:        pvc-0bebabba-a01a-11e8-81ab-000d3a377cde
Labels:        app=jenkins
               chart=jenkins-0.16.1
               heritage=Tiller
               release=jenkins
Annotations:   pv.kubernetes.io/bind-completed=yes
               pv.kubernetes.io/bound-by-controller=yes
               volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/azure-disk
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      20Gi
Access Modes:  RWO
Events:        <none>
```

**Comment by andyzhangx** [comment]
@mrdfuse Following config will take disk mount a lot more time Since gid & uid is mounted as 0(root) by default, if set as non-root(e.g. 1000), k8s will use chown/chmod to change all dir/files under that disk, this is a time consuming job, whch would make mount device very slow, in this issue: [Timeout expired waiting for volumes to attach](https://github.com/kubernetes/kubernetes/issues/67014#issuecomment-413546283), it costs about 10 min for chown/chmod operation complete.
```
      securityContext:
        runAsUser: 10000
        fsGroup: 10000 
```

**Comment by mrdfuse** [comment]
Ok that's interesting to know! We have the same problem on mounts without that config though, we just learned to live with it now.

**Comment by Korijn** [comment]
I understand that this issue has been marked stale (because it is) but I want to let people here know that they are not alone; we are seeing this issue as well.

**Comment by mrdfuse** [comment]
Good to know, we are also still experiencing very unreliable behaviour in mounting volumes. Sometimes no problem, sometimes it takes 30 minutes (also to remount existing ones)

**Comment by andyzhangx** [comment]
@mrdfuse 30 mins is totally unacceptable, could you open a new issue under https://github.com/Azure/aks-engine, and provide details like:
1. k8s version
2. azure region
3. kubectl describe po POD-NAME
4. how many disks mouting into one VM in parallel
5. vm av set or vmss

**Comment by andyzhangx** [comment]
BTW, for all disk related issues on Azure, just loop me in. And pls provide info as detailed as possible.

**Comment by trdavidson** [comment]
Hi @andyzhangx , experiencing some disc mounting issues as well (among others). Any idea what the following might entail?

```2019-04-25 09:46:23+00:00 [Warning] pod has unbound immediate PersistentVolumeClaims (repeated 4 times)```

**Comment by andyzhangx** [comment]
@trdavidson that means your pod has `unbound` PVC, you may run `kubectl describe pvc PVC-NAME` to check pvc error details.

**Comment by 27Bslash6** [comment]
Data point: We're seeing the same random volume mount timeouts on GKE v1.12.7-gke.7
